Octokit/Http/HttpClientAdapter.cs

```diff
@@ -152,5 +149,102 @@ static string GetContentMediaType(HttpContent httpContent)
             }
             return null;
         }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                if (_http != null) _http.Dispose();
+            }
+        }
+    }
+
+    internal class RedirectHandler : DelegatingHandler
+    {
+        public const string RedirectCountKey = "RedirectCount";
+        public bool Enabled { get; set; }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var response = await base.SendAsync(request, cancellationToken);
+
+            // Can't redirect without somewhere to redirect too.  Throw?
+            if (response.Headers.Location == null) return response;
+
+            // Don't redirect if we exceed max number of redirects
+            var redirectCount = 0;
+            if (request.Properties.Keys.Contains(RedirectCountKey))
+            {
+                redirectCount = (int)request.Properties[RedirectCountKey];
```

Comment:
I think this would only fail if someone pre-filled the property with a value that cannot be cast to an int.  A value is assigned to the property when it is initially created, right after this If ends.